zetaclient/chains/base/observer.go

```diff
@@ -121,18 +121,36 @@ func NewObserver(
 	return &ob, nil
 }
 
+// Start starts the observer. Returns true if the observer was already started (noop).
+func (ob *Observer) Start() bool {
+	ob.mu.Lock()
+	defer ob.Mu().Unlock()
+
+	// noop
+	if ob.started {
+		return true
+	}
+
+	ob.started = true
+
+	return false
+}
+
 // Stop notifies all goroutines to stop and closes the database.
 func (ob *Observer) Stop() {
-	ob.logger.Chain.Info().Msgf("observer is stopping for chain %d", ob.Chain().ChainId)
+	ob.logger.Chain.Info().Msgf("Stopping observer for chain %d", ob.Chain().ChainId)
 	close(ob.stop)
```

Comment:
move `close(ob.stop)` to the line after acquiring the lock? The `Stop()` is called by both
https://github.com/zeta-chain/node/blob/feat%2Fzetaclient-runtime-chains/cmd/zetaclientd/start.go#L334
and
https://github.com/zeta-chain/node/blob/feat%2Fzetaclient-runtime-chains/zetaclient/zetacore/client.go#L369

calling `close(ob.stop)` twice (the upgrade plan monitor and listener to system signals) will panic even though the chance is low.

Reply:
https://github.com/zeta-chain/node/pull/2497/commits/cc5f6530f18f26bb26cabfd0428c74f9ff7e31c9